gradle-palantir-java-format/src/test/groovy/com/palantir/javaformat/gradle/SupportsSpotless622.groovy

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+package com.palantir.javaformat.gradle
+
+import nebula.test.IntegrationTestKitSpec
+
+/**
+ * When we were getting gradle-baseline to support the configuration cache, spotless had some poorly written tasks
+ * which caused issues with the configuration cache.
+ *
+ * Bumping spotless to 6.22.0 fixed this, but revealed a new error — the palantirJavaFormat configuration was
+ * being resolved eagerly:
+ * https://github.com/palantir/palantir-java-format/blob/b7b5995df3be690780939c0d0cb2ec49b99c68c8/gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/spotless/NativePalantirJavaFormatStep.java#L45
+
+ * gradle-consistent-versions enforces against resolving configurations at configuration time, and throws an error.
+ *
+ * This test forces creation of the spotless steps, which will reveal any eager resolution of configurations.
+ */
+class SupportsSpotless622 extends IntegrationTestKitSpec {
+    private static final CLASSPATH_FILE = new File("build/impl.classpath").absolutePath
+
+    private GradlewExecutor executor
+
+    def setup() {
+        definePluginOutsideOfPluginBlock = true
+        keepFiles = true
+        executor = new GradlewExecutor(projectDir)
+    }
+
+    def "PalantirJavaFormatPlugin works with spotless 6.22.0"() {
+        // language=Gradle
+        buildFile << """
+        buildscript {
+            repositories {
+                mavenCentral() { metadataSources { mavenPom(); ignoreGradleMetadataRedirection() } }
+                gradlePluginPortal() { metadataSources { mavenPom(); ignoreGradleMetadataRedirection() } }
+            }
+             dependencies {
+                 classpath 'com.palantir.gradle.consistentversions:gradle-consistent-versions:2.34.0'
+                 classpath 'com.diffplug.spotless:spotless-plugin-gradle:6.22.0'
+             }
+        }
+
+        apply plugin: 'java'
+        apply plugin: 'com.palantir.java-format'
+        apply plugin: 'com.palantir.consistent-versions'
+        apply plugin: 'com.diffplug.spotless'
+
+        version = '0.1.0'
+        """.stripIndent(true)
+
+
+        file("versions.props")
+        file("versions.lock")
+
+        runTasks('wrapper')
+
+        buildFile << """
+            dependencies {
+                palantirJavaFormat files(file("${CLASSPATH_FILE}").text.split(':'))
+            }
+
+            // This forces the realization of the spotlessJava task, creating the spotless steps. 
+            // If any configurations are eagerly resolved in the spotless steps, 
+            // consistent-versions should catch it and throw here. 
+            project.getTasks().getByName("spotlessJava")
+        """.stripIndent()
+
+        when:
+        def result = executor.runGradlewTasks('classes', '--info')
+
+        then:
+        assert result.success
+        println(result.standardOutput)
+        result.standardOutput.contains('BUILD SUCCESSFUL')
```

Comment:
you shouldn't need to do this

Reply:
Yep, the `assert result.success` should be sufficient.